lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -708,6 +751,7 @@ def generate_adapter_map(ip = nil, gw = nil)
     end
 
     adapter_map = RbVmomi::VIM.CustomizationAdapterMapping
+    adapter_map.macAddress = mac if !mac.nil? && (mac != 'auto')
```

Comment:
Would it be possible to simplify this conditional?

`!mac.nil?` should reduce to `mac` (as `false` really isn't an option here). It also may be possible to say

`adapter_map.macAddress = mac unless mac == 'auto'` because it's already `nil`:

```Ruby
irb(main):002:0> adapter_map = RbVmomi::VIM.CustomizationAdapterMapping
=> #<RbVmomi::VIM::CustomizationAdapterMapping:0x007fb70d372f28 @props={}>
irb(main):003:0> adapter_map.macAddress
=> nil
```